### PR TITLE
Update to sqltoolsservice 5.0.0.4 and remove specific Linux distros

### PR DESCRIPTION
**Pull request by maxvolk**
Currently, the most recent Ubuntu version is 16. This PR adds those LTS versions up to the most recently released one (22).

### CODE REVIEW

**Comment by maxvolk** [comment]
@microsoft-github-policy-service agree

**Comment by Charles-Gagnon** [comment]
@kisantia @kburtram FYI - what would probably be even better is to remove the specific distros and instead move to the generic "Linux" target that Alan added support for in the downloader a while back. That will give us forward compatibility too.

e.g. what we do in MSSQL extension : https://github.com/microsoft/azuredatastudio/blob/main/extensions/mssql/config.json

**Comment by maxvolk** [comment]
@Charles-Gagnon I removed the specific distros. Also, I moved to the current release of sqltoolsservice.

File: src/configurations/config.json [hunk 1]
```diff
@@ -1,50 +1,21 @@
...
+      "Linux_ARM64": "linux-arm64-net8.0.tar.gz"
     },
     "installDir": "../sqltoolsservice/{#version#}/{#platform#}",
     "executableFiles": [
       "MicrosoftSqlToolsServiceLayer.exe",
       "MicrosoftSqlToolsServiceLayer",
       "MicrosoftSqlToolsServiceLayer.dll"
     ]
-  },
-  "v1Service": {
```
Comment (Charles-Gagnon) [inline src/configurations/config.json:28]: This is still used for downlevel Mac versions so we should keep it as-is for now

https://github.com/Microsoft/vscode-mssql/blob/79da38b0e928ade250a846a204977519e4f120cf/src/languageservice/serviceclient.ts#L233

**Comment by Charles-Gagnon** [comment]
Thanks for going through and updating all this @maxvolk! To confirm - is this a blocking issue for you on Ubuntu 22 right now?

**Comment by maxvolk** [comment]
> Thanks for going through and updating all this @maxvolk! To confirm - is this a blocking issue for you on Ubuntu 22 right now?

This issue is not blocking me on Ubuntu right now, it stems more from curiosity.

File: src/configurations/config.json (hunk 1 above)
Comment (maxvolk) [reply src/configurations/config.json:28]: What would be the rule or trade-off when one could remove such support for these old Mac OS versions? Or is in general the goal to offer maximum support to older versions?

Comment (Charles-Gagnon) [reply src/configurations/config.json:28]: We've never really set a goal or deprecation timeline, but that's definitely something we should work on! For now though continuing to support all previous versions is something we strive to adhere to.

**Comment by Charles-Gagnon** [comment]
Gotcha, thanks! Since it's not blocking we might take a bit to get to this due to some other higher priority stuff, but this is definitely work we'll want to take so we'll get to it once we have some time! If this becomes a higher priority issue for any reason let us know and I can see what we can do.

**Comment by Benjin** [comment]
I moved this linux distro consolidation work into #18179 so we don't lose it, but can also close this PR since it's been sitting around for a while

**Pull request closed without merging**